Automattic/wp-calypso
# Replace Jetpack plan checks with feature checks in some components

### Changes proposed in this Pull Request

It's not uncommon for components in Calypso to render different things based on the features included in a site plan. A couple of components in the codebase do the checks by looking at the plans. It makes adding new plans difficult, since it requires to update these components as well. This PR replaces this approach by checking the features instead (it doesn't tackle all components in the codebase).

Sequel of #45959 and #45925

### Implementation notes

I'm not sure about the exact ripple effects of adding news WPcom plans (enterprise and vip) in the plans list. The plans checks are therefore not fully replaced, but at least adding new plans won't required us to update these components, just to add features in the plans list.

### Testing instructions

- Download the PR
- Run `yarn test-client client/post-editor/editor-drawer/test/utils.js` and check all tests pass
- Run `yarn test-client client/my-sites/site-settings/seo-settings/test/utils.js` and check all tests pass
- Run `yarn test-client client/my-sites/site-settings/seo-settings/test/form.jsx` and check all tests pass
- Run `yarn test-client client/components/seo-preview-pane/test/utils.js` and check all tests pass
- Run `yarn test-client client/my-sites/site-settings/test/utils.js` and check all tests pass
